drl.go

```diff
@@ -27,17 +27,15 @@ type DRL struct {
 	currentTokenValue int64
 	open              atomic.Value
 	stopC             chan struct{}
+	once              sync.Once
 }
 
 func (d *DRL) Ready() bool {
 	return d.IsOpen()
 }
 
 func (d *DRL) IsOpen() bool {
-	if v := d.open.Load(); v != nil {
```

Comment:
Aren't you just made it less safe? Basically now it will cause panic if you call `IsOpen` before first call of `SetCurrentTokenValue`.

Reply:
nil.(bool) == false

Reply:
Maybe you wanted to comment on somewhere else, I don't even see a link between SetCurrentTokenValue and IsOpen, d.open; The atomic value (d.open) will return nil, and the (bool) cast will convert that to false.

Reply:
Interesting, some black magic which I was not aware about.

Reply:
Hm. I think I have to fix something under storage... :D

Init() sets this to true, so it's not likely to cause a panic, but I did improve the function.

https://go.dev/play/p/F2f2ybO0L_h

Reply:
Further improvement (all we needed to add was the `_` for checking if the `.(bool)` cast worked).

https://go.dev/play/p/U9euUEnTiAm